app/models/configuration_profile.rb

```diff
@@ -28,6 +29,13 @@ def activate!
     state_handler.activate!
   end
 
+  def check_ongoing_mappings
+    return unless standards_organizations.any? {|dso| dso.mappings.any?(&:in_progress?) }
```

Comment:
This N+1 query could be avoided by restructuring the scopes. I'd suggest the following configuration:

```ruby
class User
  has_many :mappings, dependent: :destroy
end

class Organization
  has_many :mappings, through: :users # instead of the `mappings` method
end

class ConfigurationProfile
  has_many :mappings, through: :standards_organizations
end
```

Then the check will look like this: `return if mappings.in_progress.none?`.